packages/vite/src/web-fonts.ts

```diff
@@ -0,0 +1,37 @@
+import type { Plugin } from 'vite'
+import { type PreflightContext, type Preset, type UnocssPluginContext, resolvePreset } from '@unocss/core'
+import { useLocalFont } from '../../preset-web-fonts/src/local-font'
+import { resolveDownloadDir } from '../../preset-web-fonts/src/util'
+import { getRemoteFontsCSS } from '../../preset-web-fonts/src/remote-font'
+
+// eslint-disable-next-line node/prefer-global/process
+const isNode = typeof process !== 'undefined' && process.stdout && !process.versions.deno
+
+export function createWebFontPlugins(ctx: UnocssPluginContext): Plugin[] {
+  return [
+    {
+      name: `unocss:web-fonts-local:dev`,
+      enforce: 'pre',
+      apply: 'serve',
```

Comment:
Remove apply, we also need generateBundle if output folder not publicDir not assetsDir: vite won't bundle css and fonts if not using any of those folders

Reply:
Let Anthony check this PR, maybe he can provide some hint